src/components/GoalDependency/GoalDependency.tsx

```diff
@@ -0,0 +1,150 @@
+import { FC, useMemo, useState } from 'react';
+import styled from 'styled-components';
+import { AutoCompleteRadioGroup, CheckboxInput, nullable } from '@taskany/bricks';
+import { gapS } from '@taskany/colors';
+import { Goal } from '@prisma/client';
+
+import { trpc } from '../../utils/trpcClient';
+import { ToggleGoalDependency, dependencyKind } from '../../schema/goal';
+import { FilterBase } from '../FilterBase/FilterBase';
+import { FilterAutoCompleteInput } from '../FilterAutoCompleteInput/FilterAutoCompleteInput';
+import { CustomCell, GoalListItemCompact } from '../GoalListItemCompact';
+import { Title } from '../Table';
+import { UserGroup } from '../UserGroup';
+
+import { tr } from './GoalDependency.i18n';
+
+const StyledAutoCompleteRadioGroup = styled(AutoCompleteRadioGroup)`
+    margin: ${gapS} 0;
+`;
+
+const StyledCheckboxInput = styled(CheckboxInput)`
+    margin: 0;
+`;
+
+const StyledGoalListItemCompact = styled(GoalListItemCompact)`
+    cursor: pointer;
+`;
+
+interface GoalDependencyProps {
+    goalId: string;
+    items: { kind: dependencyKind; goals: Goal[] }[];
+    onSubmit: (values: ToggleGoalDependency) => void;
+}
+
+const getId = (item: { id: string }) => item.id;
+const prepareGoal = (kind: dependencyKind) => (goal: Goal) => ({ id: `${kind}_${getId(goal)}`, kind, goal });
```

Comment:
I'm not sure I understand whats happening here

Reply:
Goals can be repeated for different dependencies, and when the dependency changes in radio group, react sees that the keys do not change and does not re-render the lists, which leads to incorrect behavior of displaying and selecting goals.

prepareGoal function adds a new id with the type of dependency, which solves the problem described above.